### ISSUE TITLE
gradio server recursion error triggered by gradio client

### ISSUE THREAD

**Issue by pseudotensor** [gradio-app/gradio#3906]
### Describe the bug

```
    encoded_value = jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 148, in jsonable_encoder
    if isinstance(obj, classes_tuple):
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/abc.py", line 119, in __instancecheck__
    return _abc_instancecheck(cls, instance)
RecursionError: maximum recursion depth exceeded in comparison
```
Only recursion error I've seen mentioned is: https://github.com/gradio-app/gradio/issues/1431

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Just using old gradio client 0.0.8 works fine for exactly same server and client code.

The code is not yet open-sourced, but I'll update issue if I can create a MRE.

client
```

from gradio_client import Client
client = Client(host)

def app_client_basic():
    instruction = "Who are you?"
    args = [instruction]

    api_name = '/submit'
    res = client.predict(
        *tuple(args),
        api_name=api_name,
    )
    print(md_to_text(res))

app_client_basic()
```

### Logs

```shell
This share link expires in 72 hours. For free permanent hosting and GPU upgrades (NEW!), check out Spaces: https://huggingface.co/spaces
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 429, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/applications.py", line 276, in __call__
    await super().__call__(scope, receive, send)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/starlette/middleware/cors.py", line 84, in __call__
    await self.app(scope, receive, send)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/routing.py", line 255, in app
    content = await serialize_response(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/routing.py", line 152, in serialize_response
    return jsonable_encoder(response_content)
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 131, in jsonable_encoder
    jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 161, in jsonable_encoder
    return jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 161, in jsonable_encoder
    return jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 131, in jsonable_encoder

...

  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 131, in jsonable_encoder
    jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 161, in jsonable_encoder
    return jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/site-packages/fastapi/encoders.py", line 148, in jsonable_encoder
    if isinstance(obj, classes_tuple):
  File "/home/jon/miniconda3/envs/alpaca/lib/python3.10/abc.py", line 119, in __instancecheck__
    return _abc_instancecheck(cls, instance)
RecursionError: maximum recursion depth exceeded in comparison
Keyboard interruption in main thread... closing server.
Killing tunnel 0.0.0.0:7860 <> https://cb3c9872ed571ab557.gradio.live

```
```


### System Info

```shell
gradio==3.27.0
gradio_client==0.1.3
chrome (but not relevant, just client call)
```

### Severity

blocking upgrade to latest gradio version

**Comment by freddyaboulton** [comment]
Sorry about that @pseudotensor ! But glad you figured it out.

**Comment by pseudotensor** [comment]
I actually still see it sometimes, both so far not fatal.

**Comment by pseudotensor** [comment]
Let me re-open since still occurs.  We have life HF spaces at: https://huggingface.co/spaces/h2oai/h2ogpt-chatbot2

And in logs you will see: 
```
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 161, in jsonable_encoder
    return jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 131, in jsonable_encoder
    jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 161, in jsonable_encoder
    return jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 161, in jsonable_encoder
    return jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 131, in jsonable_encoder
    jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 161, in jsonable_encoder
    return jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 117, in jsonable_encoder
    encoded_value = jsonable_encoder(
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/site-packages/fastapi/encoders.py", line 148, in jsonable_encoder
    if isinstance(obj, classes_tuple):
  File "/home/user/.pyenv/versions/3.8.9/lib/python3.8/abc.py", line 98, in __instancecheck__
    return _abc_instancecheck(cls, instance)
RecursionError: maximum recursion depth exceeded in comparison
```

The space is public, so full source code is public.  The way to trigger that issue is also public:
https://github.com/h2oai/h2ogpt/blob/fe1b1ae63d267c4a6021505b2f82c94247fe6c58/client_test.py#L20-L26

i.e. just hit that space with:
```
from gradio_client import Client
client = Client("https://h2oai-h2ogpt-chatbot2.hf.space")
args = ['', '', '', False, 'human_bot', 0.1, 0.75, 40, 1, 50, 0, False, 20, 1.0, 1, True, False, 'Who are you?', '']
res = client.predict(*tuple(args), api_name='/submit_nochat')
print(res)
```

then despite giving back a response fine, the logs are spammed with a massive recursion message.

**Comment by pseudotensor** [comment]
@freddyaboulton Thanks for checking on this!  While non-fatal, it seems like something is seriously wrong.

**Comment by freddyaboulton** [comment]
Ok - will take a look!

**Comment by freddyaboulton** [comment]
Despite the space being public, I won’t be able to see the logs unless I’m in the same org. Can you provide the full log snippet or the part that shows which gradio line is causing the stack trace? Thanks!

**Comment by pseudotensor** [comment]
Hi @freddyaboulton , I shared above partial logs.

I don't see option to download logs from spaces.  But I can copy paste for now.

[spaces.log](https://github.com/gradio-app/gradio/files/11315352/spaces.log)

**Comment by pseudotensor** [comment]
I get my problem even when I hit the server with a random api_name.  So it's not related specifically to a specific click task for me.  i.e. this causes same problem:

```
from gradio_client import Client
client = Client("https://h2oai-h2ogpt-chatbot2.hf.space")
args = []
res = client.predict(*tuple(args), api_name='/foobar')
print(res)
```

**Comment by freddyaboulton** [comment]
I'm not sure what's happening - I do see the infinite recursion error but from what I can tell that only happened once and I'm not sure if it's triggered by the client. 

After hitting the `/submit_nochat` endpoint, I just see `Using pad_token, but it is not set yet.` being printed.

![h2ogpt](https://user-images.githubusercontent.com/41651716/234337308-91e06148-c444-4026-8819-690cfa62082d.gif)

**Comment by pseudotensor** [comment]
Yes, you can ignore `Using pad_token, but it is not set yet.` -- that is just torch issue with our code.

**Comment by pseudotensor** [comment]
@freddyaboulton  any word?

**Comment by freddyaboulton** [comment]
Hi @pseudotensor !

I cloned your space and submitted three requests via the client. I see the "infinite recursion" but only when I created the client. Sending requests does not trigger that error. In this screenshot you can see that infinite recursion error but only once and you can see the "Using pad token" three times, which makes me think it's not triggered per request.

<img width="1480" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/2bd656de-c032-4268-a8ba-1f1f8436a093">

Will look at what could be causing this when the client connects (maybe something related to the `view_api` info) but seems low priority given that the app and client predictions are working well

**Comment by Turkish** [comment]
@pseudotensor  I have the same issue as this, very annoying, I get it even without the client, just by opening the /config page on a browser. What did you do to solve this issue ?

**Comment by j-dominguez9** [comment]
I'm also running into this issue. I suspect it may have something to do with a logging handler, but I don't know where to even begin fixing the issue.

Not sure if the error output is helpful, but here it is:

`  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/api.py", line 115, in post
    return request("post", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 768, in urlopen
    conn = self._get_conn(timeout=pool_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 294, in _get_conn
    return conn or self._new_conn()
                   ^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 1051, in _new_conn
    log.debug(
  File "/usr/lib64/python3.12/logging/__init__.py", line 1527, in debug
    self._log(DEBUG, msg, args, **kwargs)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1684, in _log
    self.handle(record)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1700, in handle
    self.callHandlers(record)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1762, in callHandlers
    hdlr.handle(record)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1028, in handle
    self.emit(record)
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/src/aiavatar/utils.py", line 48, in emit
    response = requests.post(self.url, json=log_entry, headers=headers)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/api.py", line 115, in post
    return request("post", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 768, in urlopen
    conn = self._get_conn(timeout=pool_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 294, in _get_conn
    return conn or self._new_conn()
                   ^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 1051, in _new_conn
    log.debug(
  File "/usr/lib64/python3.12/logging/__init__.py", line 1527, in debug
    self._log(DEBUG, msg, args, **kwargs)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1684, in _log
    self.handle(record)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1700, in handle
    self.callHandlers(record)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1762, in callHandlers
    hdlr.handle(record)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1028, in handle
    self.emit(record)
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/src/aiavatar/utils.py", line 48, in emit
    response = requests.post(self.url, json=log_entry, headers=headers)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/api.py", line 115, in post
    return request("post", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 768, in urlopen
    conn = self._get_conn(timeout=pool_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 294, in _get_conn
    return conn or self._new_conn()
                   ^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 1051, in _new_conn
    log.debug(
  File "/usr/lib64/python3.12/logging/__init__.py", line 1527, in debug
    self._log(DEBUG, msg, args, **kwargs)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1684, in _log
    self.handle(record)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1700, in handle
    self.callHandlers(record)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1762, in callHandlers
    hdlr.handle(record)
  File "/usr/lib64/python3.12/logging/__init__.py", line 1028, in handle
    self.emit(record)
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/src/aiavatar/utils.py", line 48, in emit
    response = requests.post(self.url, json=log_entry, headers=headers)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/api.py", line 115, in post
    return request("post", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/adapters.py", line 633, in send
    conn = self.get_connection_with_tls_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/requests/adapters.py", line 489, in get_connection_with_tls_context
    conn = self.poolmanager.connection_from_host(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/poolmanager.py", line 303, in connection_from_host
    return self.connection_from_context(request_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/poolmanager.py", line 328, in connection_from_context
    return self.connection_from_pool_key(pool_key, request_context=request_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/poolmanager.py", line 351, in connection_from_pool_key
    pool = self._new_pool(scheme, host, port, request_context=request_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/poolmanager.py", line 265, in _new_pool
    return pool_cls(host, port, **request_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 1006, in __init__
    super().__init__(
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 191, in __init__
    ConnectionPool.__init__(self, host, port)
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/urllib3/connectionpool.py", line 85, in __init__
    self.host = _normalize_host(host, scheme=self.scheme)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded
^CTraceback (most recent call last):
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/app/gradio_app.py", line 93, in <module>
    demo.launch(share=False)
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/gradio/blocks.py", line 2368, in launch
    ) = http_server.start_server(
        ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/gradio/http_server.py", line 149, in start_server
    server.run_in_thread()
  File "/home/joaquin/Documents/DL.ai/aiavatar-gh/.venv/lib64/python3.12/site-packages/gradio/http_server.py", line 57, in run_in_thread
    time.sleep(1e-3)
KeyboardInterrupt`
